### PR TITLE
Add expandable string enum to template2

### PR DESCRIPTION
**Pull request by olydis**
replay of https://github.com/Azure/autorest/pull/2505

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -195,8 +195,8 @@
     </pluginManagement>
   </build>
   <modules>
-    <module>test/vanilla</module>
```
Comment (RikkiGibson) [inline pom.xml]: This is a change from autorest.java that you probably want but it just got trampled in the replay.

**Comment by RikkiGibson** [comment]
This is ready to work with and fix up for merging if you `npm install -g autorest@prerelease`.

**Comment by daschult** [comment]
I've replaced this pull request with [this one](https://github.com/Azure/autorest.java/pull/10) so that I can do my work in autorest.java instead of autorest.

**Comment by olydis** [comment]
I assume this one can be closed then? Reopen if that assumption is wrong.

**Pull request closed without merging**